zitadel/v2/domain/funcs.go

```diff
@@ -25,9 +26,25 @@ func delete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Dia
 	if err != nil {
 		return diag.FromErr(err)
 	}
+	domainName := d.Id()
+	if d.Get(isPrimaryVar).(bool) {
+		resp, err := client.ListOrgDomains(ctx, &management.ListOrgDomainsRequest{})
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		for _, domain := range resp.Result {
+			parts := strings.Split(clientinfo.Domain, ":")
+			if domain.IsVerified && domain.DomainName != domainName && strings.HasSuffix(domain.GetDomainName(), parts[0]) {
```

Comment:
What's the idea behind the suffix check?

Reply:
default domain always has the suffix of the external domain

Reply:
Don't they always have the suffix of the instances primary domain which can change?
https://zitadel.com/docs/apis/resources/system/system-service-set-primary-domain

Reply:
discussed